### PR TITLE
JSTL-AdminEmail

### PR DESCRIPTION
**Pull request by taniach**

### CODE REVIEW

**Comment by taniach** [comment]
@bamboo3250 ready for review

File: src/main/java/teammates/common/datatransfer/AdminEmailAttributes.java [hunk 1]
```diff
@@ -153,4 +154,20 @@ public String getCreateDateForDisplay(){
     public String getContentForDisplay(){
         return StringHelper.recoverFromSanitizedText(this.getContent().getValue());
     }
+    
+    public String getFirstAddressReceiver(){
+        return this.addressReceiver.get(0);
+    }
+    
+    public boolean getHasGroupReceiver() {
+        return this.getGroupReceiver() != null && this.getGroupReceiver().size() > 0;
```
Comment (bamboo3250) [inline src/main/java/teammates/common/datatransfer/AdminEmailAttributes.java]: Perhaps `groupReceiver` can be a better name?

File: src/main/webapp/WEB-INF/tags/admin/email/sentEmail.tag [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<%@ tag description="sent.tag - Sent email row" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@ attribute name="sentEmail" required="true" type="teammates.ui.template.AdminSentEmailRow" %>
+
+<tr id="${sentEmail.emailId}">
+    <td>
+        <a <c:forEach items="${sentEmail.actions.editButton.attributes}" var="attribute">
+           ${attribute.key}="${attribute.value}"
```
Comment (bamboo3250) [inline src/main/webapp/WEB-INF/tags/admin/email/sentEmail.tag]: please indent this line

File: src/main/webapp/WEB-INF/tags/admin/email/sentEmail.tag [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+           </c:forEach>>
+               ${sentEmail.actions.editButton.content}
+        </a>
+        
+        <a <c:forEach items="${sentEmail.actions.deleteButton.attributes}" var="attribute">
+           ${attribute.key}="${attribute.value}"
```
Comment (bamboo3250) [inline src/main/webapp/WEB-INF/tags/admin/email/sentEmail.tag]: and this one.

File: src/main/webapp/WEB-INF/tags/admin/email/trash.tag [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                        <c:otherwise>
+                            No Trash Email
+                        </c:otherwise>
+                    </c:choose>
+                </span>
+            </strong>
+            <%-- Empty trash button --%>
+            <span class="pull-right">
+                <a <c:forEach items="${trashEmailTable.emptyTrashButton.attributes}" var="attribute">
+                   ${attribute.key}="${attribute.value}"
```
Comment (bamboo3250) [inline src/main/webapp/WEB-INF/tags/admin/email/trash.tag]: this line as well

File: src/main/webapp/WEB-INF/tags/admin/email/trashEmail.tag [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<%@ tag description="trash.tag - Trash email row" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@ attribute name="trashEmail" required="true" type="teammates.ui.template.AdminTrashEmailRow" %>
+
+<tr id="${trashEmail.emailId}">
+    <td>
+        <a <c:forEach items="${trashEmail.actions.editButton.attributes}" var="attribute">
+           ${attribute.key}="${attribute.value}"
```
Comment (bamboo3250) [inline src/main/webapp/WEB-INF/tags/admin/email/trashEmail.tag]: and this

File: src/main/webapp/WEB-INF/tags/admin/email/trashEmail.tag [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+           </c:forEach>>
+               ${trashEmail.actions.editButton.content}
+        </a>
+        
+        <a <c:forEach items="${trashEmail.actions.moveOutOfTrashButton.attributes}" var="attribute">
+           ${attribute.key}="${attribute.value}"
```
Comment (bamboo3250) [inline src/main/webapp/WEB-INF/tags/admin/email/trashEmail.tag]: hopefully this is the last one :sweat_smile:

File: src/main/java/teammates/common/datatransfer/AdminEmailAttributes.java (hunk 1 above)
Comment (taniach) [inline src/main/java/teammates/common/datatransfer/AdminEmailAttributes.java]: The method name has to start with 'get' or 'is' in JSTL

**Comment by taniach** [comment]
@bamboo3250 ready for review

File: src/main/java/teammates/common/datatransfer/AdminEmailAttributes.java (hunk 1 above)
Comment (bamboo3250) [inline src/main/java/teammates/common/datatransfer/AdminEmailAttributes.java]: Don't know why I thought it was an attribute. Do we really need this function? We can just use `not empty groupReceiver` in JSTL, right?

**Pull request merged**